packages/core/admin/admin/src/content-manager/pages/EditView/index.js

```diff
@@ -29,62 +28,44 @@ import Informations from './Informations';
 import Header from './Header';
 import { createAttributesLayout, getFieldsActionMatchingPermissions } from './utils';
 import DeleteLink from './DeleteLink';
+import GridRow from './GridRow';
 
 const cmPermissions = permissions.contentManager;
 const ctbPermissions = [{ action: 'plugin::content-type-builder.read', subject: null }];
 
 /* eslint-disable  react/no-array-index-key */
-const EditView = ({
-  allowedActions,
-  isSingleType,
-  goBack,
-  layout,
-  slug,
-  id,
-  origin,
-  userPermissions,
-}) => {
+const EditView = ({ allowedActions, isSingleType, goBack, slug, id, origin, userPermissions }) => {
   const { trackUsage } = useTracking();
   const { formatMessage } = useIntl();
   const { createActionAllowedFields, readActionAllowedFields, updateActionAllowedFields } =
-    useMemo(() => {
-      return getFieldsActionMatchingPermissions(userPermissions, slug);
-    }, [userPermissions, slug]);
+    getFieldsActionMatchingPermissions(userPermissions, slug);
+
+  const { layout, formattedContentTypeLayout } = useSelector((state) => {
+    const layout = state['content-manager_editViewLayoutManager'].currentLayout;
+
+    return {
+      layout,
+      formattedContentTypeLayout: createAttributesLayout(layout?.contentType ?? {}),
+    };
+  });
```

Comment:
I really like the idea to use a selector here 💯. Even more since it means removing complex proptypes.

I'm not sure about the impact on performance though. I saw the comment from @joshuaellis:

>why don't we instead select this layout state in the EditView component and return that as well as doing this calculation and let redux handle the memoization?

But looking at the redux [docs for deriving state](https://redux.js.org/usage/deriving-data-selectors), it seems to me that Redux will run the selector function for every Redux action. My understanding is that the memoization they do is only to check if the component needs to rerender based on the result of the selector. Which would mean for us to re-run the expensive `createAttributesLayout` on every action.

And also since we are returning a new object, I think the component will always re-render since the object identity will be new every time.

So we seem to be in these 2 situations mentioned in the docs:

<img width="983" alt="Screenshot 2022-11-22 at 11 01 15" src="https://user-images.githubusercontent.com/8087692/203284980-c272c394-fd52-48bb-a711-eaad8f28d1e3.png">

Their suggestion is to use reselect, which we already have in Strapi, to separate the actual selector and the function that derives the data. What do you think about it?

Reply:
Great catch @remidej! 

I think if they're suggesting reselect lets definitely use it, also bonus it's already in the repo ⭐